src/libraries/System.Private.CoreLib/src/System/Collections/Concurrent/ConcurrentQueueSegment.cs

```diff
@@ -45,7 +46,7 @@ internal ConcurrentQueueSegment(int boundedLength)
         {
             // Validate the length
             Debug.Assert(boundedLength >= 2, $"Must be >= 2, got {boundedLength}");
-            Debug.Assert((boundedLength & (boundedLength - 1)) == 0, $"Must be a power of 2, got {boundedLength}");
+            Debug.Assert(BitOperations.IsPow2(boundedLength), $"Must be a power of 2, got {boundedLength}");
```

Comment:
Do you think its worth an analyzer to help recognize common cases like these and point users (and ourselves) to using the `BitOperations` (or with Generic Math, directly using `T.IsPow2`/etc)?

Many of these are pretty common bit twiddling patterns and are cases where users may not know we have a "better" accelerated version available.

Reply:
Yeah, I think that'd be useful. My assumption is it won't really help with perf (e.g. our IsPow2 doesn't appear to be better), but it does help with code clarity, and if we improve ours in the future, any use will benefit (though presumably the jit could also recognize more twiddling patterns in the future, as it does now for blsr).

Reply:
I also wonder how do you hunt these cases. I did some regex searches in my previous PR to find such updatable patterns.

Reply:
regex :-)